### ISSUE TITLE
Impact Hammer

### ISSUE THREAD

**Issue by HellDragon-HK** [GreatEmerald/UT3Weapons#11]
- ANIMATION RELATED
 - [ ] 3rd person does not start animating until the sound is on it's 2nd loop and the hammer has already been charged for a few seconds, also it goes straight to the loop animation with no start which looks odd.

- GAME PLAY RELATED
 - [ ] Self damage inconsistent with UT3
 - [ ] 95% sure alt fire does not disarm powerups, very hard to test as the AI in DM doesn't want to grab the UDamage for some reason in 1on1 and they are very accurate on any difficulty against an approaching hammer.
 - [ ] Hammer's forward damage radius might be too small, I had no trouble using it in UT3 but in UT2004 it's very hard to use, it's like the target needs to be part way into the gun before it auto fires

 - SOUND RELATED
  - [ ] Primary fire loop often goes silent until death after a few uses or weapon switch, sometimes comes back when held which skips the start up sound and goes straight to the loop
  - [ ] Alt fire first use start up sound isn't stopped by firing if fired pre-loop start point
  - [ ] Self damage jumping has a specific sound called BodyLaunch in UT3 editor, unsure if it's used but the editor says it is 
  - [ ] Impact sounds for wall hits are not currently used
  - [ ] Impact sounds for vehicle hits are not currently used
  - [ ] Charging loop doesn't match fire start volume and they should be exactly the same so it's something in the code

-VISUAL REALTED
 - [ ] UT2004 muzzle flash is erm wimpy I guess you could say compared to the quick blue explosion I see on UT3 which goes across the entire screen nearly
 - [ ] Alt fire has a bright blue muzzle flash with electricity bolts on UT3, on UT2004 the primary and alt has the same muzzle flash

**Comment by HellDragon-HK** [comment]
Shine isn't missing on 32 bit so it's another 64 bit issue....get the feeling I'm going to be adding that 'works with 32 bit' text a lot now and already saw it instantly on the enforcer as well.

**Comment by GreatEmerald** [comment]
I think the self-damage works quite differently in UT2004 than UT3. IIRC UT2004 depends on the distance to surface or something. Or maybe it doesn't depend on duration... I also made sure that you can't shieldjump while jumping.

**Comment by HellDragon-HK** [comment]
Should I test impact hammer self damage then between UT3 and UT2004 since I can see both? I'm currently on DM-Arsenal and DM-Idoma so it would be easy to do

**Comment by GreatEmerald** [comment]
Sure.

**Comment by HellDragon-HK** [comment]
Uhh this can't be right....UT3 100 hp to 75 hp and in UT2004 100 hp to 96 hp and both were full charged shots

**Comment by GreatEmerald** [comment]
How about uncharged shots?

**Comment by HellDragon-HK** [comment]
Also bizarre, UT3 100 to 93 and UT2004 100 to 96 again, button was only tapped no charge made
Does surface type matter because UT2004 is currently wood and UT3 is I believe metal
Should probably try some other things like blowing myself up with a rocket now I guess huh

**Comment by GreatEmerald** [comment]
The surface shouldn't matter as far as I know.
Hm, I think I see the bug! I forgot to set SelfDamageScale. If the damage should be 25, given that it deals 140 damage max normally, it should be set to 0.121.
Should be fixed in https://github.com/GreatEmerald/UT3Weapons/commit/9cbebf5ad47fa8c29d74975a66d386a065caa4de .

**Comment by HellDragon-HK** [comment]
I've never used git pull for UT3Weapons, if you mean for me to try it out can you tell me the path I need?

**Comment by GreatEmerald** [comment]
I don't think you have it cloned yet. You should run `git clone https://github.com/GreatEmerald/UT3Weapons.git` in the UT2004 folder.
If it doesn't compile like UT3Vehicles, you can open another issue here.

**Comment by HellDragon-HK** [comment]
I did git clone but I'm not seeing a UT3Weapons folder anywhere
EDIT-Wait I found it, it's in my Users folder and not in UT2004 and I also see UT3Vehicles so they are not going in my UT folder, for now I'll just move it where it needs to go but how do I fix this issue in the future?

**Comment by GreatEmerald** [comment]
Use `cd` to go to the right directory before you clone...

**Comment by HellDragon-HK** [comment]
ok so it's cd path git clone path then?
EDIT-(So as to not make a meaningless comment to the message below) Thank you for confirmation

**Comment by GreatEmerald** [comment]
Yes.

**Comment by HellDragon-HK** [comment]
Impact Hammer primary full charge still not doing much
In UT3 I get 75 in UT2004 I get 88 from 100
UT2004 took 9 erm hits I guess you could say to die and UT3 only takes 4

**Comment by GreatEmerald** [comment]
Hm, that is odd, the maths must be wrong somewhere... Or the damage could be relative to the distance or something like that. I'll need to check.

**Comment by HellDragon-HK** [comment]
I tested on different maps so surface type can be ruled out and I also aimed different distances in UT2004 and got the same result but give it a shot

**Comment by HellDragon-HK** [comment]
Impact Hammer in UT3 does 0 damage to vehicles and can cause self damage (on a Manta i was instantly killed from single impact hammer full charge), Impact Hammer also doesn't push most vehicles in UT3....also Impact is auto firing on vehicles when in range it doesn't do this in UT3

In UT2004 Impact did 28-29 damage to Manta and gave it medium to massive knockback with no self damage, Scorpion took 29 damage and very little knockback with no self damage

EDIT - In UT3 Impact does little knockback to Raptor and some self damage, in UT2004 it does between little and fair (collision might be interfering with how much exactly) knockback with no self damage

In UT3 the Impact causes little to fair knockback and self damage (odd it actually is more effective than every weapon in throwing the Cicada around with the exception of the AVRiL, haven't tried a rocket yet though), and in UT2004 it does cause about the same knockback but without the self damage and it also auto fires

**Comment by GreatEmerald** [comment]
Yea, autofire and more knockback is intentional. The Impact Hammer is supposed to be effective against Manta's crushing ability.

**Comment by HellDragon-HK** [comment]
Altfire doesn't seem work on powerups, I'm not 100% sure because UT2004 AI is dead serious about not letting you get close enough to do it, as soon as you start charging it they turn and fire with high accuracy  on any difficulty setting, I'm able to use the impact hammer easily in UT3 so what's different here?  It feels like in UT3 the hammer might have a bit more auto fire radius, shield gun is also easier to use than this.

This is something I wasn't ever aware of but does the alt fire work like an EMP mine in UT3 and disable vehicles?

EDIT - Impact has better much larger (too large) hit radius against vehicles than against players.

EDIT2 - Discovered animation issue
